cmd/di.go

```diff
@@ -938,3 +950,19 @@ func (di *Dependencies) handleConnStateChange() error {
 		latestState = e.State
 	})
 }
+
+func (di *Dependencies) bootstrapResidentCountry() error {
+	if di.EventBus == nil {
+		return errMissingDependency("di.EventBus")
+	}
+
+	if di.LocationResolver == nil {
+		return errMissingDependency("di.LocationResolver")
+	}
+	di.ResidentCountry = identity.NewResidentCountry(di.EventBus, di.LocationResolver)
+	return nil
+}
+
+func errMissingDependency(dep string) error {
```

Comment:
Idea is nice and cool and I dont generally disagree with this, but wont it get too messy if we start doing this for every dependency?

Reply:
it's already pretty messy. The DI part is rather hard to comprehend, a major refactor is in order :)